KILTprotocol/demo-client
# feat: added QR Code including service address

## Fixes KILTProtocol/ticket#316
* Added a QR code displaying the public identity including a `serviceAddress` to the identity card (visible under `/dashboard`).
* Added a minimal "encoding" setup (simply an array), for the QR Code to be readable by mobile cameras.

## How to test:

- Checkout the branch and run locally alongside the node and the services
- Create a new identity e.g. with the mnemonic "dog hen effort cradle win embark tuition need fly stumble short damage"
- You should see a field Public Identity as below. By default the QR Code should be hidden. 
- Click the button "Show". The QR Code should be displayed and the button label verb should be "Hide".
- Click the button again to hide.

<img width="1047" alt="Screenshot 2020-02-18 at 12 08 32" src="https://user-images.githubusercontent.com/9762897/74731389-2af45500-5248-11ea-9065-46aabf4e3e06.png">
<img width="1062" alt="Screenshot 2020-02-18 at 12 08 38" src="https://user-images.githubusercontent.com/9762897/74731378-2760ce00-5248-11ea-99d5-77f10a8a4f5b.png">


- Scan the QR Code with a mobile phone
- Your phone should read the QR Code and display the following data:
`["5EwY7eCYAAcLwANHv8s18m5LnYnPyJM8PhvaWWqkeJq4QVtu","0xdbcea99a89c26039f6b05d8d458fafa308524ca1382b7b5f8a3b6482a871a864","//127.0.0.1:3000/messaging"]`

![IMG_0416](https://user-images.githubusercontent.com/9762897/74722987-1c06a600-523a-11ea-9057-64b3120f7725.PNG)
![IMG_0417](https://user-images.githubusercontent.com/9762897/74723000-21fc8700-523a-11ea-960d-972459c42b72.PNG)

NB: follow-up tickets: 
* https://github.com/KILTprotocol/ticket/issues/317
* https://github.com/KILTprotocol/ticket/issues/318


## Checklist:
